packages/peregrine/lib/talons/RootComponents/Category/useCategory.js

```diff
@@ -0,0 +1,143 @@
+import { useEffect, useMemo, useRef } from 'react';
```

Comment:
This is a bit of scope creep, but this RootComponent didn't have a talon. I broke it up without changing anything about the logic.

Reply:
Careful - I'm fairly sure @eug123 is doing this already https://github.com/magento/pwa-studio/pull/2686/files#diff-c08d9f0ea0b6c1b26fb3111ffffeadd8